From 34f55eac4652d49913f25ff592b6189953887bf1 Mon Sep 17 00:00:00 2001
From: Jianxu Zheng <jian.xu.zheng@intel.com>
Date: Thu, 19 Mar 2015 15:30:01 +0800
Subject: [PATCH 313/314] atomisp2: for isp2401 depth mode will use copy pipe

for depth mode ov680 stream, running in yuvpp pipe would cause
a fps drop and has to run in copy pipe.

Change-Id: Ibe7a9bf02064cc5bb046d7accfc1b56e4c0ca473
Tracked-On: #JIMINAN-26401
Signed-off-by: Jianxu Zheng <jian.xu.zheng@intel.com>
Signed-off-by: Tao Wei <wei.tao@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/46791
---
 .../camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c    | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index ac3477b..8d83664 100755
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -4847,7 +4847,9 @@ static void atomisp_check_copy_mode(struct atomisp_sub_device *asd,
 	    (!atomisp_subdev_format_conversion(asd, source_pad))) ||
 	    ((asd->isp->inputs[asd->input_curr].type == SOC_CAMERA) &&
 	    (asd->isp->inputs[asd->input_curr].camera_caps->
-		sensor[asd->sensor_curr].stream_num > 1)))
+		sensor[asd->sensor_curr].stream_num > 1)) ||
+	    ((asd->depth_mode->val) &&
+	     (!atomisp_subdev_format_conversion(asd, source_pad))))
 		asd->copy_mode = true;
 	else
 #endif
-- 
1.9.1

